### DISCUSSION TITLE
Possible to throw when input has a field that is not expected?

### DISCUSSION THREAD

**Discussion by notflip** [colinhacks/zod#1641] [Q&A] [answered]
Is it possible to validate the input data, so that when a field is present that is not expected, we get an error?

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#strict

**Reply by notflip** [reply]
@JacobWeisenburger Thanks for your answer, it seems like this only checks 1 level deep in my z.object(), should strict() also check deeper down in nested objects?

**Reply by JacobWeisenburger** [reply]
As far as I know there is no `deepStrict()` or anything like that. I think the way `strict()` works right now is correct, because it allows for you to opt in only to the objects that need it.

**Reply by notflip** [reply]
Okay that makes sense! Thanks Jacob

**Reply by JacobWeisenburger** [reply]
Zod is always open to PRs, so feel free to contribute a solution to this.
